### ISSUE TITLE
unable to decrypt after closing app

### ISSUE THREAD

**Issue by diadal** [ajimae/ncrypt-js#5]
once you close the app and reopen it decrypt stop working

**Comment by ajimae** [comment]
Hi @diadal 

Can you provide more context on this issue?
E.g what do specifically mean by closing and reopening the app?

If possible please provide an example or a code snippet let's see how we can resolve this issue.

Thanks

**Comment by diadal** [comment]
Android and iOS App

**Comment by ajimae** [comment]
Okay, I will try and reproduce this ASAP then circle back to give updates. 🥂 
Thank you.

**Comment by ajimae** [comment]
@diadal kindly checkout this commit #6 and see if it fixes your issue.

cheers🥂

**Comment by ajimae** [comment]
Closing this due to inactivity, if issue persists, feel free to reopen.
Cheers🥂

**Comment by laztheripper** [comment]
I have this issue as well. secret is a static string.

```
const ncrypt = require('ncrypt-js');
const { encrypt, decrypt } = new ncrypt(config.api.secret);

var s = 'somestr';
var es = encrypt(s);
var ds = decrypt(es);

console.log(s);
console.log(es);
console.log(ds);
```

running my node app
```
somestr
5818308c467f3c2919fbc2970e2c1060.95bf4f24b60615ce271d20d56daaf152858dff1ec1517a9fdd3508909f0e384a
somestr
```

stopping it, and starting it again
```
somestr
3e43be34b1431e731d07c8f989d4dfc9.ea277318e240792e067476c305128b2483002de1936fc3af2f2fa9a0c28ffd4e
somestr
```

Thanks

**Comment by diadal** [comment]
use this https://github.com/diadal/secure-ls

**Comment by ajimae** [comment]
Hello @laztheripper 

I can't reproduce this from my end as everything works perfectly.
@diadal had issues with an earlier version (=< 1.1.0) of this library which is now deprecated.

I tested it with this code snippet by running the `node app.js` hit the `/encrypt` endpoint.

Stopped the server, and also use the encrypted string from the `/encrypt` endpoint call as the decrypt payload and everything works just fine.


```js
// app.js
var express = require('express');
var Ncrypt = require('ncrypt-js');

var server = express();
var { encrypt, decrypt } = new Ncrypt('shhh its a secret');

server.use(express.json());
server.use(express.urlencoded({ extended: true }));

server.post('/encrypt', function(req, res) {
  var { data } = req.body;
  var en = encrypt(data);

  res.status(200).json({
    encryptedData: en
  });
});

server.post('/decrypt', function(req, res) {
  var { data } = req.body;
  var de = decrypt(data);

  res.status(200).json({
    originalData: de
  });
});

server.listen(3000, function() {
  console.log('server running...');
});
```


```js
POST http://localhost:3000/encrypt
Content-Type: application/json

{
  "data": "somestr"
}

###

POST http://localhost:3000/decrypt
Content-Type: application/json

{
  "data": "a5b2108c795a9e6a51a08166ec69dabe.3b4ea804bbf9f02e6043c9b12a1dc66064b3424b91e65cc52719a58636173221"
}

```

Test out the above snippet and feel free to stop the node app between encrypt and/or decrypt everything should work fine🥂.

**Comment by ajimae** [comment]
@laztheripper if you could also help, can you post the error you are getting when trying to decrypt?

Thanks

**Comment by diadal** [comment]
@ajimae once you reload the page or your app session change you cant decrypt your data anymore

**Comment by ajimae** [comment]
@diadal update your version of the library

Thanks 🥂

**Comment by laztheripper** [comment]
> @laztheripper if you could also help, can you post the error you are getting when trying to decrypt?
> 
> Thanks

I had the latest version, there was no error, it simply didn't return the same result between opening and closing the node process. If the node process was kept up, I could encrypt and decrypt any number of requests, but the moment I restarted the app it wouldn't produce the same hash. I'm not saying it didn't encrypt and decrypt, it did, but when I encrypt() the same string twice, I expected the same output hash every time. I read up a bit about this and it turns out that's a difference between deterministic and probabilistic encryption (which this is the latter I suppose). I switched to bcrypt and locked all the variables to produce the deterministic hashes I wanted, although it errors on some secret strings, but if you test one and it works, it'll always work. Thanks for looking at this issue anyway!
